src/common/components/repository-row/index.js

```diff
@@ -201,26 +202,26 @@ export class RepositoryRowView extends Component {
     }
 
     const nextSnapcraftData = nextProps.snap.snapcraftData;
-    const currentSnapcraftData = this.props.snap.snapcraftData;
-    const snapcraftNameOwnership = nextSnapcraftData && nextProps.nameOwnership[nextSnapcraftData.name];
 
-    // name ownership status is available when we have it or is already fetching
-    const isNameOwnershipAvailable = (snapcraftNameOwnership &&
-      (snapcraftNameOwnership.nameOwnershipStatus || snapcraftNameOwnership.isFetching)
+    if (snapNameIsMismatched(nextProps.snap)) {
+      this.checkNameOwnership(nextProps.nameOwnership, nextSnapcraftData.name);
+    }
+
+    if (nextProps.snap.storeName) {
+      this.checkNameOwnership(nextProps.nameOwnership, nextProps.snap.storeName);
```

Comment:
The main 'functional' change in this PR - if snap has registered name, check its ownership.